training/CHAPEL_TUTORIAL.md

```diff
@@ -993,358 +1072,618 @@ Expected Output:
 
-<a id="ex22"></a>
-### Example 22: Variable Declarations
-Throughout this tutorial we've mostly used `var` for our variable declarations,
-but there are some instances where we used `const`. This begs the question, what are the different ways to declare
-variables and when should you use them?
+<a id="locale"></a>
+### Locales
+We've mentioned `Locale`s briefly in earlier sections, but let's dive into them a bit deeper.
+I like to think of a locale as a single computer in a cluster of computers all working together to solve a problem.
+This isn't always the case, but it is a useful model. More generally a
+[locale](https://chapel-lang.org/docs/users-guide/locality/localesInChapel.html) is
+"a piece of a target architecture that has processing and storage capabilities".
 
-[`const`](https://chapel-lang.org/docs/users-guide/base/constParam.html) is used when a variable shouldn't ever be changed.
-This is a program execution time constant, so the compiler doesn't need to know its value.
-Knowing that this value won't change can help the compiler to make optimizations.
 
-[`param`](https://chapel-lang.org/docs/users-guide/base/constParam.html#declaring-params) is very similar to `const`,
-but it's a program compilation time constant. So it's value does need to be known by the compiler.
+For any given computer in our cluster, accessing the data it stores will be faster than having it fetch data
+from a different computer in the cluster. This is because the data will need to be transferred from the computer
+that has it to the computer that needs it. 
 
-[`ref`](https://chapel-lang.org/docs/language/spec/variables.html?highlight=ref#ref-variables) is used to avoid creating a
-copy of an array or repeated accesses of an attribute (especially within a loop).
-Keep in mind changes to the `ref` will update the variable it is referencing.
+Data stored on a computer is "local" to that computer and data stored on
+a different computer is "remote". This terminology reflects the time penalty we have to pay when accessing "remote" data when compared
+to "local" data. This same intuition holds for locales in general.
 
-```chapel
-// pretend myBool is determined during runtime
-var myBool = true;
+The big takeaway is it's in our best interest to minimize how often locales need to operate
+on remote data.
 
-proc helper(myBool: bool) {
-    return if myBool then 5 else 10;
-}
+To reiterate a bit more exactly:
 
-// use a var if you expect a value to change
-var myVar = [0, 1, 2];
-// we use a const because we don't know the value at compilation time
-const myConst = helper(myBool);
-// we use a param becasue we know what the value is at compilation time
-param myParam = 17;
+> Say `x` and `y` are both stored on `locale_i`. From the perspective of `locale_i`, we say `x` and `y` are
+both local. There no significant difference in access times.
+> 
+> But if `x` is on `locale_i` and `y` is on `locale_j`. From the perspective of `locale_i`, we say `x` is local
```

Comment:
I think it would be helpful to also add the third case of `y` being local and `x` being remote. Not sure if it is too far of a leap at this point to make a reference to how these are used in Source and Destination Aggregators, or at least maybe link the sections.

Reply:
I appreciate the feedback! It certainly wouldn't be hard to update, but I think it'd just be duplicating the second case but with ther variable names swapped

like in my head the three possible cases are:
- both local
- one local, one remote
- both remote

so since we chose `x` and `y` arbitrarily, I think these both fall under the second

I def see why you jumped to copy aggregation since you're already primed to think about that. But here we're just defining the terms local and remote

Reply:
That makes sense to me